packages/cloudflare-optimizer-scripts/test/index.test.js

```diff
@@ -105,11 +105,36 @@ describe('handleEvent', () => {
     expect(output).toBe(`transformed-${input}`);
   });
 
+  it('should handle redirects opaquely', async () => {
+    const mockedResponse = new Response('', {
+      status: 302,
+      statusText: '',
+    });
+
+    const event = {
+      request: {method: 'GET', url: 'https://example.com/'},
+      respondWith: jest.fn(),
+      passThroughOnException: jest.fn(),
+    };
+
+    global.fetch.mockReturnValue(mockedResponse);
+    handleEvent(event, defaultConfig);
+
+    expect(global.fetch).toBeCalledTimes(1);
```

Comment:
nit: testing implementation details is more brittle than testing function output of `respondWith`. In this case you are already checking for referentially equal returned value of the output. Does testing the number of times fetch was called or that it included a manual redirect directive increase confidence? Can we test for those features in an externally visible way instead?

Reply:
Thank you for detailed review. I have made some changes and removed the brittle fetch part.

I think as long as event.respondWith is responding with a same mocked redirect response we are good.